hbase-server/src/test/java/org/apache/hadoop/hbase/util/TestRegionMover.java

```diff
@@ -238,4 +245,139 @@ public void testTargetServerDeadWhenLoading() throws Exception {
       assertFalse(rm.load());
     }
   }
+
+  @Test
+  public void testDecomServerExclusionWithAck() throws Exception {
+    MiniHBaseCluster cluster = TEST_UTIL.getHBaseCluster();
+    HRegionServer excludeServer = cluster.getRegionServer(1);
+    List<HRegion> regions = excludeServer.getRegions();
+    int regionsExcludeServer = excludeServer.getNumberOfOnlineRegions();
+    TEST_UTIL.getAdmin().decommissionRegionServers(
+      Collections.singletonList(excludeServer.getServerName()), false);
+
+    waitForServerDecom(excludeServer);
+
+    HRegionServer regionServer = cluster.getRegionServer(0);
+    String rsName = regionServer.getServerName().getHostname();
+    int port = regionServer.getServerName().getPort();
+    String hostname = rsName + ":" + Integer.toString(port);
+    RegionMoverBuilder rmBuilder = new RegionMoverBuilder(hostname, TEST_UTIL.getConfiguration())
+      .ack(true);
+
+    int targetServerRegions = cluster.getRegionServer(2).getRegions().size();
+    int sourceServerRegions = regionServer.getRegions().size();
+
+    try (RegionMover regionMover = rmBuilder.build()) {
+      Assert.assertTrue(regionMover.unload());
+      LOG.info("Unloading {}", hostname);
+      assertEquals(0, regionServer.getNumberOfOnlineRegions());
+      assertEquals(regionsExcludeServer, cluster.getRegionServer(1).getNumberOfOnlineRegions());
+      LOG.info("Before:" + regionsExcludeServer + " After:" +
+        cluster.getRegionServer(1).getNumberOfOnlineRegions());
+      List<HRegion> regionList = cluster.getRegionServer(1).getRegions();
+      int index = 0;
+      for (HRegion hRegion : regionList) {
+        Assert.assertEquals(hRegion, regions.get(index++));
+      }
+      Assert.assertEquals(targetServerRegions + sourceServerRegions,
+        cluster.getRegionServer(2).getNumberOfOnlineRegions());
+      Assert.assertTrue(regionMover.load());
+    }
+
+    TEST_UTIL.getAdmin().recommissionRegionServer(excludeServer.getServerName(),
+      Collections.emptyList());
+  }
+
+  private void waitForServerDecom(HRegionServer excludeServer) throws IOException {
+
+    TEST_UTIL.getAdmin().decommissionRegionServers(
+      Collections.singletonList(excludeServer.getServerName()), false);
+
+    TEST_UTIL.waitFor(3000, () -> {
```

Comment:
is this 3s? If so, is this enough?

Reply:
Actually, 3s is too much :) Upper bound ideally should not be more than 1s, this is just to be on safer side.